quickannotator/api/v1/image/routes.py

```diff
@@ -50,29 +55,187 @@ def delete(self, args):
+                func.ST_AsGeoJSON(db_models.Image.embedding_coord).label('embedding_coord')
+            ).filter(db_models.Image.project_id == project_id).all()
+            if images is not None:
+                return images, 200
+            else:
+                abort(404, message="Image not found")
 #################################################################################
+WSI_extensions = ['svs', 'tif','dcm','vms', 'vmu', 'ndpi',
+                  'scn', 'mrxs','tiff','svslide','bif','czi']
+JSON_extensions = ['json','geojson']
+
+def import_geojson_annotation_file(image_id, annotation_class_id, gtpred, filepath):
+    '''
+    This is expected to be a geojson feature collection file, with each polygon being a feature.
+    
+    '''
+    # path = filepath.split("quickannotator/")[1]
+    with open(filepath, 'r') as file:
+        # Load the JSON data into a Python dictionary
+        data = json.load(file)["features"]
+    all_anno = []
+    
+    table_name = f"{image_id}_{annotation_class_id}_{gtpred}_annotation"
+    
+    
+    for i, d in enumerate(tqdm(data)):
+
+        shapely_geometry = shape(d['geometry'])
+        annotation = {
+            "image_id": image_id,
+            "annotation_class_id": annotation_class_id,
+            "isgt": gtpred == "gt",
+            "centroid": shapely_geometry.centroid.wkt,
+            "area": shapely_geometry.area,
+            "polygon": shapely_geometry.wkt,
+            "custom_metrics": json.dumps({"iou": 0.5}) 
+        }
+        
+        all_anno.append(annotation)
+        
+        if len(all_anno)==1_000:
+            table = Table(table_name, Base.metadata, autoload_with=engine)
+            stmt = insert(table).values(all_anno)
+            db_session.execute(stmt)
+            db_session.commit()
+            all_anno = []
+    
+    # commit any remaining annotations
+
+    table = Table(table_name, Base.metadata, autoload_with=engine)
+    stmt = insert(table).values(all_anno)
+    db_session.execute(stmt)
+    db_session.commit()
+
+def create_annotation_table(image_id, annotation_class_id, gtpred):
+    table_name = f"{image_id}_{annotation_class_id}_{gtpred}_annotation"
+    table = models.Annotation.__table__.to_metadata(Base.metadata, name=table_name)
+    Base.metadata.create_all(bind=engine, tables=[table])
+
+@bp.route('/upload/file', methods=["POST"])
+def upload_files():
+    """Upload a file"""
+    
+    file = request.files["file"]
+    project_id = request.form.get('project_id')
+    print('upload_files ~~~~~~~~~')
+    print(project_id)
+    if file and project_id:
+        filename = secure_filename(file.filename)
+
+        # get file extension
+        file_basename, file_ext = os.path.splitext(filename)
+        file_ext = file_ext[1:]
+        # handle image file
+        if file_ext in WSI_extensions:
+            full_project_path = os.path.join(current_app.root_path, projects_path, f'proj_{project_id}/images/temp')
```

Comment:
FYI, in my export annotations PR I introduce a new class called FileSystemManager.
https://github.com/choosehappy/QuickAnnotator/blob/b7179bd63fd6f0937bfc127a0be61a8f701da125/quickannotator/db/utils.py#L21

This will be used to abstract away from having to build paths manually. If we decide to change our filesystem structure, the interface will stay the same, e.g.,

```python
fsman = FileSystemManager() # gets default configuration e.g., BASE_PATH from constants.py
path = fsman.get_project_image_path(project_id, image_id)
```